doc/reference/storage_powerflex.md

```diff
@@ -0,0 +1,151 @@
+The VMs root volume contains the `.b` block suffix whilst an ISO file ends on `.i`.
+
+In order to have enough space for the snapshot identifier, at least two more characters need to be
+reserved. You can try keeping the snapshots name at a minimum by setting [`snapshots.pattern`](storage-powerflex-snapshot-pattern)
+to `%d`. This will result in the following volume names:
+
+* Snapshot `snap0` for a virtual machine `v1`: `v_project1_v1@0`
+* Snapshot `snap1` for a custom volume `vol`: `u_project1_vol@1`
+
+(storage-powerflex-limitations)=
+### Limitations
+
+The `powerflex` driver has the following limitations:
+
+Limit of snapshots in a single vTree
+: From any volume in PowerFlex only 126 snapshots can be taken due to an internal limitation in the PowerFlex vTree.
+  This limit also applies to any child of any of the parent volumes snapshots.
+  A single vTree can only have 126 branches.
+
+Non optimized image storage
+: Due to the limit of 126 snapshots in the vTree, the PowerFlex driver doesn't come with support for optimized image storage.
+  This would limit LXD to create only 126 instances from an image.
+  Instead, when launching a new instance the images contents get copied to the instances root volume.
+
+Copying volumes
+: PowerFlex does not support creating a copy of the volume so that it gets its own vTree.
+  Therefore, LXD falls back to copying the volume on the local system.
+  This implicates an increased use of bandwidth due to the volumes contents being transferred over the network twice.
+
+Volume size constraints
+: In PowerFlex the size of a volume needs to be in multiples of 8 GiB.
+  This results in the smallest possible volume size of 8 GiB.
+  However, if not specified otherwise, volumes are getting thin-provisioned by LXD.
+  PowerFlex volumes can only be increased in size.
+
+Volume name size limitation
+: In PowerFlex the name of a volume cannot exceed 31 characters.
+  LXD uses certain identifiers to indicate the purpose of a volume in the storage backend.
+  See the section on [Volume names](storage-powerflex-volume-names) for more information.
+
+Sharing custom volumes between instances
+: The PowerFlex driver "simulates" volumes with content type `filesystem` by putting a file system on top of a PowerFlex volume.
+  Therefore, custom storage volumes can only be assigned to a single instance at a time.
+
+Sharing the PowerFlex storage pool between installations
+: Sharing the same PowerFlex storage pool between multiple LXD installations is not supported.
+  Due to the volume name limitations, no extra identifier can be added to distinguish between installations.
+
+## Configuration options
+
+The following configuration options are available for storage pools that use the `powerflex` driver and for storage volumes in these pools.
+
+(storage-powerflex-pool-config)=
+### Storage pool configuration
+
+Key                        | Type   | Default             | Description
```

Comment:
Do we expect this PR to go in before or after https://github.com/canonical/lxd/pull/12645 ?
Either way, these options will need to be moved to the code eventually (which also gives us the option to add more information to some of the options).
I'll just do a quick review now since we'll need to fix it again later anyways.

Reply:
The options should also be ordered alphabetically, but again, this will come for free with the other PR, so no need to fix now.

Reply:
Now that https://github.com/canonical/lxd/pull/12645 got merged I rebased from `main` and moved the config keys for volumes and pools into the code.